subprojects/kotlin-dsl-integ-tests/src/integTest/kotlin/org/gradle/kotlin/dsl/integration/PrecompiledScriptPluginIntegrationTest.kt

```diff
@@ -721,4 +723,71 @@ class PrecompiledScriptPluginIntegrationTest : AbstractPluginIntegrationTest() {
             assertThat(output, containsString("my-plugin settings plugin applied"))
         }
     }
+
+    @Test
+    fun `should not allow precompiled plugin to conflict with core plugin`() {
+        withKotlinBuildSrc()
+        withFile(
+            "buildSrc/src/main/kotlin/java.gradle.kts",
+            """
+            tasks.register("myTask") {}
+            """
+        )
+        withDefaultSettings()
+        withFile(
+            "build.gradle",
+            """
+            plugins {
+                java
+            }
+            """
+        )
+
+        val error = buildAndFail("help")
+
+        error.assertHasCause(
+            "Precompiled plugin: '${Paths.get("src", "main", "kotlin", "java.gradle.kts")}' conflicts with the core plugin: 'java'.\n\n"
```

Comment:
Is the file separator different in this message depending on Linux/Windows? 

You could send this through `TextUtil.normaliseFileSeparators` to have a stable string, I think.

Reply:
Yeah, platform is the issue :). Thanks for suggestion, I will try `TextUtil.normaliseFileSeparators`.